lib/ansible/modules/storage/glusterfs/glusterfs_volume.py

```diff
@@ -0,0 +1,408 @@
+                     cluster.
+    bricks:
+        required: True (If and action is in [create, add-brick, remove-brick])
+        description: Specifies the bricks that constitutes the volume if
+                     the action is volume create, the new bricks to be
+                     added to a volume if the action is volume add-brick
+                     and the bricks to be removed out of a volume
+                     if the action is volume remove-brick. Format of each
+                     brick should be hostname:brick_path.
+    force:
+        required: False
+        description: Specifies if an operation is to be forced.
+                     Applicable only to peer detach, volume { create, start,
+                     stop, add-brick}
+    key:
+        required: True if action is set
+        description: Set action sets options for the volume. Key
+                     specifies which option is to be set
+    value:
+        required: True if action is set
+        description: Specifies the value to be set for the option
+                     specified by key.
+    transport:
+        required: False
+        description: Specifies which transport type is to be used
+                     while creating the volume. Default is tcp.
+    replica:
+        required: False
+        description: Specifies the volume to be created is of type
+                     replicate or not. Can also be used to change the
+                     type while adding or removing bricks
+    replica_count:
+        required: True if replica is yes
+        description: Specifies the replica count if the volume type is
+                     to be replica
+    arbiter_count:
+        required: False
+        description: If volume is of type replica, arbiter_count will
+                     specify the number of bricks that will be configured
+                     as arbiter node. If replica_count is 3 and arbiter_count
+                     is 1, 3rd brick of the replica is automatically
+                     configured as an arbiter node.
+    disperse:
+        required: False
+        choices: [yes, no]
+        description: Specifies if the volume is to be disperse or not
+    disperse_count:
+        required: False
+        description: Specifies the number of disperse bricks. If not
+        specified, the number of bricks specified is taken as the
+        <count> value.
+    redundancy_count:
+        required: False
+        description: Specifies the number of redundancy bricks. If
+        if not specified and the volume if of type disperse,
+        it's default value is computed so that it generates an
+        optimal configuration.
+    state:
+        required: True
+        choices: ["absent", "present", "remove-brick", "started",
+                  "stopped", "replace-brick", "rebalance",
+                  "add-brick", "profile", "set", "barrier", "bitrot" ]
```

Comment:
state = barrier, bitrot, replace-brick, remove-brick etc seems like a wrong usage of state. For instance, there can be bitrot state as started/present but bitrot itself is not a state of a gluster volume.